ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildingFactory.service.ts

```diff
@@ -277,18 +278,18 @@ export class BuildingFactoryService {
     return answers;
   }
 
-  private getMatrix(selectedAnswers: Object, questDef: Object): any {
-    return Object.entries(selectedAnswers['matrixSelected']).map(([questionId, answerId]): Object => ({
-        question: this.getMatrixQuestion(questionId, questDef['rows']),
-        answer: this.getMatrixAnswer((answerId as string), questDef['options'])
+  private getMatrix(selectedAnswers: Object, questDef: Object): MatrixAnswer[] {
+    return Object.entries(selectedAnswers['matrixSelected']).map(([questionId, answerId]): MatrixAnswer => ({
+        verticalAnswer: this.getMatrixVerticalAnswer(questionId, questDef['rows']),
+        horizontalAnswer: this.getMatrixHorizontalAnswer((answerId as string), questDef['options']),
       }));
   }
 
-  private getMatrixQuestion(questionId: string, rows: Array<Object>): string {
-    return rows.find(question => question['rowStableId'] === questionId)['rowText'];
+  private getMatrixVerticalAnswer(questionId: string, rows: Row[]): Row {
+    return rows.find(question => question['rowStableId'] === questionId);
```

Comment:
`question.rowStableId`  is better for finding some type errors then `question['rowStableId']`.
The same below.

Reply:
Actually, it is already changed :D the same happened, as I mentioned above in comment. Thanks.